### ISSUE TITLE
Error in ApcMetricsPoller.php

### ISSUE THREAD

**Issue by jerryxh** [upwork/phystrix-dashboard#5]
The regex expression on line #100 of ApcMetricsPoller.php throws an exception if one of the following APCu entries are found:

    1) phystrix_cb_MyCommandopened
    2) phystrix_cb_MyCommandsingle_test_blocked

These are both valid entries and should not throw an exception.

To prevent this from happening, I recommend changing:

    if (!isset($match[1])) {
        throw new \RuntimeException('Invalid counter key: ' . $counter['key']);
    }

to:

    if (!isset($match[1])) {
        continue;
    }

**Comment by lcf** [comment]
but if we do that, the valid entries would be skipped. Should we fix the regular expression instead? 
or, even, remove it altogether, simply assume that it matches. Remove the prefix and use the rest as the command key. what do you think @jerryxh ?

**Comment by jerryxh** [comment]
I am not an expert in Phystrix, but if you look at the execute() method in the AbstractCommand class, you will notice that whenever APCu entries of the form phystrix_cb_MyCommandopened or phystrix_cb_MyCommandsingle_test_blocked are generated there is also a simultaneous call to increment or reset the bucket, which generates or increments an entry of the form phystrix_cb_MyCommand_1_123456789. So it seems to me that the "opened" and "single_test_blocked" entries are redundant and can be skipped because entries of the form phystrix_cb_MyCommand_1_123456789 would not be screened out by the regex expression on line #100. My understanding is that the "opened" and "single_test_blocked" entries are only used for checking if the circuit is open and if a single test is allowed, respectively. They don't seem to be used for the metrics calculations.

**Comment by lcf** [comment]
You are absolutely right @jerryxh! I believe the dashboard has been tested with normal operation only (we ended up relying on other tools for visibility more).
We need to fix this, if you care to submit a PR it would be much appreciated. The solution you proposed looks fine to me.

**Comment by jerryxh** [comment]
Yes, I will open a pull request.
